### PR TITLE
#186426833 : Proactive sync instead of Reactive sync,check block flag in governanc…

### PR DESCRIPTION
**Pull request by qmjava**
…e rule

Pull Request review Checklist:
 - [x] `Follows` guidelines for `public` repo ?
 - [x] `README` updated, if required ?
 - [x] `Release Version` updated ?
 - [x] `User Agent` Version updated ?
 - [x] `Includes` tests conducted and `test result` in the ticket and/or QA doc ?
 - [x] `Does NOT` include code from others?  (Unintended consequence of a bad merge) ?

### CODE REVIEW

File: Moesif.Middleware/Helpers/Tasks.cs [hunk 1]
```diff
@@ -58,21 +58,6 @@ static List<EventModel> QueueGetAll(ConcurrentQueue<EventModel> moesifQueue, int
                         var createBatchEventResponse = await client.Api.CreateEventsBatchAsync(batchEvents);
                         createBatchEventResponse = createBatchEventResponse.ToDictionary(k => k.Key.ToLower(), k => k.Value);
 
-                        // Signal events
-                        var configETag = createBatchEventResponse["x-moesif-config-etag"];
```
Comment (dgilling) [inline Moesif.Middleware/Helpers/Tasks.cs:62]: @qmjava , why did we get rid of checking etag? we still want to sync if etag changes? We should also ensure thread safe (dont launch a large number of syncs in parallel)

**Review by dgilling — CHANGES_REQUESTED** [review]

File: Moesif.Middleware/Helpers/Tasks.cs (hunk 1 above)
Comment (qmjava) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: Because there is a dedicated thread for sync both appConfig and governance rule.

Comment (dgilling) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: The bg thread is a failsafe. We still want to push changes as soon as we see an etag change. Thats the whole point of etag. The question earlier is why did etag not work. We even tested sending one event to ensure SDK saw updated etag on response, but still did not reflect.

Comment (qmjava) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: Well, the only issue with this approach is that there are have to be some events. If there are no events, 
they won't get synced.

Comment (dgilling) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: yes, that is why we have both. At scale, we expect etag to be dominant. However, if server is idle, we expect the timed based trigger to be dominant.

Comment (praves77) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: eTag syncs configs/rules immediately for high frequency ingestion whereas timer-based scheduler works fine for low-frequency.

Comment (qmjava) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: Well, what if there are no events?

Comment (praves77) [reply Moesif.Middleware/Helpers/Tasks.cs:62]: in that case, nothing to worry about. Scheduler will update the config/rules on its own time.

File: Moesif.Middleware/NetCore/MoesifMiddlewareNetCore.cs [hunk 2]
```diff
@@ -162,10 +162,10 @@ private void ScheduleAppConfig()
                     }
                     catch (Exception e)
                     {
-                        _logger.LogError(e, "Error while scheduling appConfig job");
+                        _logger.LogError(e, "Error while updating appConfig");
                     }
-                    // wait for max 1 hour
-                    await Task.Delay(60 * 1000);
+                    
+                    configEvent.WaitOne(60 * 1000);
```
Comment (praves77) [inline Moesif.Middleware/NetCore/MoesifMiddlewareNetCore.cs:168]: this is one minute not an hour. Correct ?

File: Moesif.Middleware/Helpers/GovernanceHelper.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ namespace Moesif.Middleware.Helpers
 {
     public class GovernanceHelper
     {
-        public static async Task<Governance> updateGovernance(MoesifApiClient client, Governance prevGovernance, bool debug, ILogger logger)
+    public static async Task<Governance> updateGovernance(MoesifApiClient client, Governance prevGovernance, bool debug, ILogger logger)
```
Comment (praves77) [inline Moesif.Middleware/Helpers/GovernanceHelper.cs:18]: any reason, indentation has been removed?

**Comment by praves77** [comment]
@qmjava Please update [Moesif.Middleware.nuspec 's version info](https://github.com/Moesif/moesif-dotnet/blob/81736c2656641bd3714ca4db73e6fa7c4ee09583/Moesif.Middleware/Moesif.Middleware.nuspec#L5) always whenever we change/bump up the version otherwise it can not be published.

File: Moesif.Middleware/NetCore/MoesifMiddlewareNetCore.cs (hunk 2 above)
Comment (qmjava) [reply Moesif.Middleware/NetCore/MoesifMiddlewareNetCore.cs:168]: Correct.

File: Moesif.Middleware/Helpers/GovernanceHelper.cs (hunk 3 above)
Comment (qmjava) [reply Moesif.Middleware/Helpers/GovernanceHelper.cs]: fat finger

**Review by praves77 — APPROVED** [review]

**Pull request merged**